### PR TITLE
[VerticalTabs] Show context menu in empty spaces of vertical tabs container.

### PR DESCRIPTION
**Pull request by jagadeshjai**
- Show context menu similar to tabstrip in empty spaces of vertical tabs container.

Resolves https://github.com/brave/brave-browser/issues/41825

# OutCome
[brave_vertical_tabs_context_menu.webm](https://github.com/user-attachments/assets/69a1a887-c55f-40ef-9c60-7418e885c157)

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. Enable vertical tab via settings page or from tab's context menu.
2. Right click on empty spaces in vertical tab
3. Now Context menu should be shown

### CODE REVIEW

**Comment by jagadeshjai** [comment]
@fallaciousreasoning Could you kindly review this PR?

File: browser/ui/views/frame/vertical_tab_strip_root_view_browsertest.cc [hunk 1]
```diff
@@ -3,22 +3,23 @@
...
 #include "brave/browser/ui/views/frame/vertical_tab_strip_region_view.h"
-#include "brave/browser/ui/views/frame/vertical_tab_strip_root_view.h"
 #include "brave/browser/ui/views/frame/vertical_tab_strip_widget_delegate_view.h"
 #include "brave/browser/ui/views/tabs/vertical_tab_utils.h"
 #include "chrome/browser/ui/views/frame/browser_non_client_frame_view.h"
 #include "chrome/browser/ui/views/frame/browser_view.h"
 #include "chrome/browser/ui/views/tabs/tab_strip.h"
 #include "chrome/test/base/in_process_browser_test.h"
 #include "content/public/test/browser_test.h"
+#include "gtest/gtest.h"
```
Comment (fallaciousreasoning) [inline browser/ui/views/frame/vertical_tab_strip_root_view_browsertest.cc:18]: nit: 

```suggestion
#include "testing/gtest/include/gtest/gtest.h"
```

**Review by fallaciousreasoning — APPROVED** [review]
lgtm % nit - going to kick off CI :smile:

File: browser/ui/views/frame/vertical_tab_strip_root_view_browsertest.cc (hunk 1 above)
Comment (jagadeshjai) [reply browser/ui/views/frame/vertical_tab_strip_root_view_browsertest.cc]: Done.

**Comment by bsclifton** [comment]
Looks like CI passed - good for merge, @fallaciousreasoning?

**Comment by fallaciousreasoning** [comment]
Sure is @bsclifton

**Comment by bsclifton** [comment]
Nice work, @jagadeshjai! 😄👍

**Comment by jagadeshjai** [comment]
Thanks for merging this PR @bsclifton !

**Pull request merged**